### DISCUSSION TITLE
When is it useful to call RUST ?

### DISCUSSION THREAD

**Discussion by LeaKitty** [fzyzcjy/flutter_rust_bridge#317] [Q&A]
I can't figure out when it's more useful to call RUST or not (but except on heavy computation), could you give me some concrete cases?

For example :
- Is it more "efficient" to call a  database written in RUST instead of Dart ([skytable](https://github.com/skytable/skytable), [lucid](https://github.com/lucid-kv/lucid) or other) ?
- On a simple "match" that tests 10 different occurrences, is it more "winning" to call RUST instead of DART ?
- Etc.

Thanks :)

**Comment by fzyzcjy** [comment]
> could you give me some concrete cases?

For my own app in production: Some computer vision algorithms. Using Rust I can play with libs like OpenCV and write complex algorithms, but with Dart, performance is much worse and no good lib.

> Is it more "efficient" to call a database written in RUST instead of Dart ([skytable](https://github.com/skytable/skytable), [lucid](https://github.com/lucid-kv/lucid) or other) ?

A db written in rust should perform better than a db written in dart, IMHO. But firstly ensure you really need that much performance.

> On a simple "match" that tests 10 different occurrences, is it more "winning" to call RUST instead of DART ?

Not sure what is a "match"... But general idea is: rust is much faster, but firstly ensure you really need that speed.
